main.go

```diff
@@ -0,0 +1,362 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"slices"
+	"strconv"
+	"strings"
+
+	"github.com/icinga/check-netgear/internal/checks"
+	"github.com/icinga/check-netgear/netgear"
+
+	"github.com/NETWAYS/go-check"
+	"github.com/NETWAYS/go-check/result"
+)
+
+// So that flag supports slices
+type stringSliceFlag []string
+
+func (s *stringSliceFlag) String() string { return strings.Join(*s, ",") }
+func (s *stringSliceFlag) Set(v string) error {
+	*s = append(*s, v)
+	return nil
+}
+
+type intSliceFlag []int
+
+func (i *intSliceFlag) String() string {
+	parts := make([]string, 0, len(*i))
+	for _, v := range *i {
+		parts = append(parts, strconv.Itoa(v))
+	}
+	return strings.Join(parts, ",")
+}
+func (i *intSliceFlag) Set(v string) error {
+	n, err := strconv.Atoi(v)
+	if err != nil {
+		return err
+	}
+	*i = append(*i, n)
+	return nil
+}
+
+// Flags contains all command line flags that are relevant to check modes
+type Flags struct {
+	NoPerfdata bool
+
+	HideCpu  bool
+	HideMem  bool
+	HideTemp bool
+	HideFans bool
+
+	CpuWarn  float64
+	CpuCrit  float64
+	MemWarn  float64
+	MemCrit  float64
+	TempWarn float64
+	TempCrit float64
+	FanWarn  float64
+	FanCrit  float64
+	PortWarn float64
+	PortCrit float64
+
+	PortsToCheck intSliceFlag
+}
+
+// ModeBasic contains all the basic hardware information of the switch, including CPU and RAM usage, temperature and fan
+// speed
+func ModeBasic(netgearSession *netgear.Netgear, flags *Flags) (*result.PartialResult, error) {
+	deviceInfo, err := netgearSession.DeviceInfo()
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving device info: %w", err)
+	}
+
+	if len(deviceInfo.DeviceInfo.Details) == 0 {
+		return nil, fmt.Errorf("error retrieving device info")
+	}
+	upTime := deviceInfo.DeviceInfo.Details[0].Uptime
+
+	o := result.PartialResult{
+		Output: fmt.Sprintf("Device Info: Uptime - %v", upTime),
+	}
+
+	if !flags.HideCpu {
+		if len(deviceInfo.DeviceInfo.Cpu) == 0 {
+			return nil, fmt.Errorf("no CPU info for this device")
+		}
+		cpuUsage, err := netgear.StringPercentToFloat(deviceInfo.DeviceInfo.Cpu[0].Usage)
+		if err != nil {
+			return nil, fmt.Errorf("error parsing CPU usage: %v\n", err)
```

Comment:
Still needs to be addressed, as I stated [yesterday](https://github.com/Icinga/check-netgear/pull/6#discussion_r2560006050).


```suggestion
			return nil, fmt.Errorf("error parsing CPU usage: %w", err)
```

Reply:
Changed it.